src/models/local-user.js

```diff
@@ -0,0 +1,12 @@
+"use strict";
+
+module.exports = class LocalUser {
+	constructor(name, role) {
```

Comment:
I was thinking about that, but the reason why I didn't is because there is a chain of things that could cause undetected harm: config contains passwords, and then user gets sent as it through the socket in the server code.
So yeah, we'd be ignoring it in the constructor, and we can even blacklist in the server code, but I can see how one day we might end up sending something to the client that shouldn't be sent to the client, and that would be horrific.

That being said, I'm just telling what I was thinking when doing that, I'm not against improving this, again, this was very early thought and I didn't want to invest time if it was getting trashed, but I'm completely happy testing stuff to reduce the future pain without making security bugs easy to make.

Reply:
> I was thinking about that, but the reason why I didn't is because there is a chain of things that could cause undetected harm

Right, sure, but we are currently just getting the stuff out of the config before this constructur, yeah. Surely it would be just as easy to do that in here. I am definitely not suggesting storing the entire config in the user (that would be bad) but the constructor takes in the config, we destructure out what we want from the config, and then we store *only those bits*. It's essentially what we are doing now, but it seems clearner to me (otherwise we will end up with this having many, many things in the constructor).

We still have to explicitly take bits out of the config, so there isn't a risk to accidentally adding something that we don't want to. Does that make sense?

Reply:
Yeah it does, and I think I have reached to the same conclusion by the time I finished my previous message :)
I still have small concerns, but I'll play with this and see if actually doing this address them, which is rather likely. I'll change, commit, and we'll probably decide that it's fine.